src/connect.test.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const connect = require('./connect');
+
+const connected1 = connect({ x: 1, y: 2 });
+const connected2 = connect({ x: 1, y: 2 });
+const connected3 = connect();
+const connected4 = connect();
+
+test('Should work correctly in case of conflict', () => {
+  expect(connected1({ z: 3, x: 10 }))
+    .toStrictEqual({ x: 10, y: 2, z: 3 });
+});
```

Comment:
describe('Basic scenario', () => {
  const baseParams = { x: 1, y: 2 };
  const extraParams = { z: 3, x: 10 };
  let connected;

  beforeEach(() => {
    connected = connect(baseParams);
  });

  test('should return expected object', () => {
    expect(connected(extraParams))
      .toStrictEqual({  x: 1, y: 2, z: 3, x: 10 });
  });

  test('should not change baseParams', () => {
    expect(baseParams)
      .toStrictEqual({ x: 1, y: 2 });
  });

  test('should not change extraParams', () => {
    expect(extraParams)
      .toStrictEqual({ z: 3, x: 10 });
  });
});